### DISCUSSION TITLE
Issue installing brew & brew packages

### DISCUSSION THREAD

**Discussion by GPHofficial** [Homebrew/discussions#184] [Everyday usage] [answered]
Is homebrew-core being rate limited ? 

Tried to brew install a package on Big Sur and Catalina (Different Nodes)

Big Sur is having a clean install and Catalina is on a previously working brew installation.

Both are stuck in the step of git cloning homebrew-core, tried setting clone depth to 1 and downloading directly from Github's download zip.

Gotten 503 on Github download zip (2) and the following error (1) during git clone of depth 1. Tried cloning other repository that succeeded. Internet speed is >500Mbps down so, it shouldn't be slow internet.

(1) error: RPC failed; curl transfer closed with outstanding read data remaining
![Screenshot 2020-11-30 at 4 16 31 PM](https://user-images.githubusercontent.com/2346151/100584755-7a16fc80-3327-11eb-84b5-0e0288e9b39e.png)

(2) https://codeload.github.com/Homebrew/homebrew-core/zip/master
![Screenshot 2020-11-30 at 4 10 28 PM](https://user-images.githubusercontent.com/2346151/100584432-1260b180-3327-11eb-9e7e-9aff69c5026c.png)

**Comment by SMillerDev** [comment, marked as answer]
You should contact GitHub support for this. Homebrew has no control over the way GitHub handles load balancing.

**Reply by GPHofficial** [reply]
Have the team considered using a CDN of sorts to speed up the updates ? Given that it runs pretty much every time there's a new install/update is needed, i'd imagine that the workload on Github end must be quite a fair bit.

How about a prepackaged version so that git wouldn't have to retrieve the entire history of 200,000 commits or possibly a shallow clone

Anyways, I've used HOMEBREW_AUTO_UPDATE_SECS to speed up my installs. So thanks for adding the flag.

**Reply by SMillerDev** [reply]
> How about a prepackaged version so that git wouldn't have to retrieve the entire history of 200,000 commits or possibly a shallow clone

If Microsoft can't manage to scale this properly as a service I don't see how 20-something volunteers are going to handle that in their free time. And the shallow clone actually makes the performance of GitHub as a platform worse so we've been asked to avoid that.
